llmfoundry/data/finetuning/tasks.py

```diff
@@ -403,8 +408,11 @@ def filter_long_or_empty_examples(example: Dict) -> bool:
         if dist.get_local_rank() == 0:
             os.remove(signal_file_path)
 
+        if error:
+            log.error('Error during data prep')
+            raise error
         log.debug('All ranks finished data prep')
-        return filtered_dataset
+        return filtered_dataset # type: ignore
```

Comment:
why type ignore?

Reply:
because filtered_dataset is of type unbound since it's declared inside the try catch statement.

Reply:
ah, maybe a bit cleaner to declare it as `None` outside of the try/except, and then just assert its a dataset instance right before return

Reply:
fixed